Update rollbacks.md (#338)
rancher/rancher-docs: docs/getting-started/installation-and-upgrade/install-upgrade-on-a-kubernetes-cluster/rollbacks.md

---
title: Rollbacks
---
## Additional Steps for Rollbacks with Rancher v2.6.4+

> In Rancher v2.6.4, the cluster-api module has been upgraded from v0.4.4 to v1.0.2 in which the apiVersion of CAPI CRDs are upgraded from `cluster.x-k8s.io/v1alpha4` to `cluster.x-k8s.io/v1beta1`. This has the effect of causing rollbacks from Rancher v2.6.4 to any previous version of Rancher v2.6.x to fail because the previous version the CRDs needed to roll back are no longer available in v1beta1.

Reviewer: ```suggestion
Rancher v2.6.4 upgrades the cluster-api module from v0.4.4 to v1.0.2. Version v1.0.2 of the cluster-api, in turn, upgrades the apiVersion of CAPI CRDs from `cluster.x-k8s.io/v1alpha4` to `cluster.x-k8s.io/v1beta1`. The `cluster.x-k8s.io` upgrade to v1beta1 causes rollbacks to fail when you attempt to move from Rancher v2.6.4 to any previous version of Rancher v2.6.x. This is because v1beta1 doesn't contain the CAPI CRD versions that the rollback is attempting to restore.
```

> To avoid this, the Rancher resource cleanup scripts should be run **before** the restore or rollback is attempted. Specifically, two scripts have been created to assist you: one to clean up the cluster (`cleanup.sh`), and one to check for any Rancher-related resources in the cluster (`verify.sh`). Details on the cleanup script can be found in the [rancher/rancher-cleanup repo](https://github.com/rancher/rancher-cleanup).

Reviewer: ```suggestion
To avoid rollback failure, the Rancher resource cleanup scripts should be run **before** the restore operation or rollback is attempted. Specifically, two scripts have been created to assist you: 

* `verify.sh`:  Checks for any Rancher-related resources in the cluster. 
*  `cleanup.sh`: Cleans up the cluster.

Details on the cleanup script can be found in the [rancher/rancher-cleanup repo](https://github.com/rancher/rancher-cleanup).
```

:::caution

> Rancher will be down as the `cleanup` script runs as it deletes the resources created by rancher.

Reviewer: ```suggestion
 There will be Rancher downtime while `cleanup.sh` runs, since the script deletes resources created by Rancher.
```

:::

> The additional preparations:

Reviewer: ```suggestion
### Rolling back from v2.6.4+ to lower versions of v2.6.x
```

1. Follow these [instructions](https://github.com/rancher/rancher-cleanup/blob/main/README.md) to run the scripts.
1. Follow these [instructions](https://rancher.com/docs/rancher/v2.6/en/backups/migrating-rancher/) to install the rancher-backup Helm chart on the existing cluster and restore the previous state.
    1. Omit Step 3.

>     1. When Step 4 is reached, install the required Rancher v2.6.x version on the local cluster you intend to roll back to.

Reviewer: ```suggestion
    1. When you reach Step 4, install the Rancher v2.6.x version on the local cluster you intend to roll back to.
```

---
title: Rollbacks
---
## Additional Steps for Rollbacks with Rancher v2.6.4+

> Rancher v2.6.4 upgrades the cluster-api module from v0.4.4 to v1.0.2. Version v1.0.2 of the cluster-api, in turn, upgrades the CAPI CRDs apiVersion from `cluster.x-k8s.io/v1alpha4` to `cluster.x-k8s.io/v1beta1`. The CAPI CRDs upgrade to v1beta1 causes rollbacks to fail when you attempt to move from Rancher v2.6.4 to any previous version of Rancher v2.6.x. This is because v1beta1 is missing the previous versions that the CRDs need to roll back.

Reviewer: ```suggestion
Rancher v2.6.4 upgrades the cluster-api module from v0.4.4 to v1.0.2. Version v1.0.2 of the cluster-api, in turn, upgrades the Cluster API's  Custom Resource Definitions (CRDs) from `cluster.x-k8s.io/v1alpha4` to `cluster.x-k8s.io/v1beta1`. The CRDs upgrade to v1beta1 causes rollbacks to fail when you attempt to move from Rancher v2.6.4 to any previous version of Rancher v2.6.x. This is because necessary components in v1alpha4 are no longer available in v1beta1.
```
